geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateJndiBindingCommand.java

```diff
@@ -0,0 +1,260 @@
+      @CliOption(key = CliStrings.IFNOTEXISTS, help = IFNOTEXISTS__HELP,
+          specifiedDefaultValue = "true", unspecifiedDefaultValue = "false") boolean ifNotExists,
+      @CliOption(key = DS_CONFIG_PROPERTIES, optionContext = "splittingRegex=,(?![^{]*\\})",
+          help = DS_CONFIG_PROPERTIES_HELP) ConfigProperty[] dsConfigProperties)
+      throws IOException, SAXException, ParserConfigurationException, TransformerException {
+
+    boolean alreadyExists = isBindingAlreadyExists(jndiName);
+    if (alreadyExists && ifNotExists)
+      return ResultBuilder.createInfoResult(CliStrings.format(
+          "Skipping creation of Jndi Binding with jndi-name \"{0}\" as its already exists.",
+          jndiName));
+    else if (alreadyExists)
+      return ResultBuilder.createUserErrorResult(
+          CliStrings.format("JNDI Binding with jndi-name \"{0}\" already exists.", jndiName));
+
+    JndiBindingConfiguration configuration = new JndiBindingConfiguration();
+    configuration.setBlockingTimeout(blockingTimeout);
+    configuration.setConnectionPoolDatasource(connectionPooledDatasource);
+    configuration.setConnectionUrl(connectionUrl);
+    configuration.setIdleTimeout(idleTimeout);
+    configuration.setInitPoolSize(initPoolSize);
+    configuration.setJdbcDriver(jdbcDriver);
+    configuration.setJndiName(jndiName);
+    configuration.setLoginTimeout(loginTimeout);
+    configuration.setManagedConnFactory(managedConnFactory);
+    configuration.setMaxPoolSize(maxPoolSize);
+    configuration.setPassword(password);
+    configuration.setTransactionType(transactionType);
+    configuration.setType(type);
+    configuration.setUsername(username);
+    configuration.setXaDatasource(xaDataSource);
+    if (dsConfigProperties != null && dsConfigProperties.length > 0)
+      configuration.setDsConfigurations(Arrays.asList(dsConfigProperties));
+
+    ClusterConfigurationService service = getSharedConfiguration();
+
+    if (service == null) {
+      return ResultBuilder.createUserErrorResult("Cluster Configuration Service is not available. "
+          + "Please connect to a locator with running Cluster Configuration Service.");
+    }
+
+    updateXml(configuration);
+
+    Set<DistributedMember> targetMembers = findMembers(null, null);
+    if (targetMembers.size() > 0) {
+      List<CliFunctionResult> jndiCreationResult =
+          execute(new CreateJndiBindingFunction(), configuration, targetMembers);
+      return buildResult(jndiCreationResult);
+    }
+
+    return ResultBuilder
+        .createInfoResult(CliStrings.format("jndi-binding \"{0}\" created.", jndiName));
+  }
+
+  List<CliFunctionResult> execute(CreateJndiBindingFunction function,
+      JndiBindingConfiguration configuration, Set<DistributedMember> targetMembers) {
+    return executeAndGetFunctionResult(function, configuration, targetMembers);
+  }
+
+  boolean isBindingAlreadyExists(String jndiName)
```

Comment:
would be a good idea to put this method in cluster config -- starting to think about making xml manipulation into the cluster config service. something like:

// find the xml element in a group's xml, with the tagname that has this attribute
getElement(String group, String tagName, String attribute, String value)

Reply:
I would wait until more than one command to use it. So I think thats a good refactoring that can be done while adding the other commands.